app/assets/javascripts/discourse-plugins/index.js

```diff
@@ -267,4 +267,57 @@ module.exports = {
       return true;
     }
   },
+
+  contentFor(type, config) {
+    if (this.shouldLoadPlugins() && type === "test-plugin-js") {
```

Comment:
```suggestion
    if (!this.shouldLoadPlugins()) {
      return;
    }

    if (type === "test-plugin-js") {
```

then the following two else-ifs could drop that check also